src/main/java/org/folio/linked/data/integration/kafka/sender/CreateMessageSender.java

```diff
@@ -1,14 +1,15 @@
 package org.folio.linked.data.integration.kafka.sender;
 
 import java.util.Collection;
-import java.util.function.Consumer;
+import java.util.function.BiConsumer;
 import java.util.function.Function;
 import org.folio.linked.data.model.entity.Resource;
 
-public interface CreateMessageSender extends Function<Resource, Collection<Resource>>, Consumer<Resource> {
+public interface CreateMessageSender
+  extends Function<Resource, Collection<Resource>>, BiConsumer<Resource, Boolean> {
 
-  default void produce(Resource resource) {
+  default void produce(Resource resource, Boolean putIndexDate) {
```

Comment:
Ok.. I see that we invoke `workCreateMessageSender.accept(work, false);` in BatchIndexServiceImpl (line 41).

Maybe it is better to do as follows so that CreateMessageSender is not impacted?

class WorkCreateMessageSender implements CreateMessageSender {

 // We will invoke this accept message from CreateMessageSender's produce method.
  @Override
  public void accept(Resource resource) {
    this.accept(resource, TRUE);
  }

  // We will invoke this "special" method from BatchIndexServiceImpl
  public void accept(Resource resource, Boolean putIndexDate) {
    var message = bibliographicSearchMessageMapper.toIndex(resource)
      .type(CREATE);
    bibliographicMessageProducer.sendMessages(List.of(message));
    if (TRUE.equals(putIndexDate)) {
      publishIndexEvent(resource);
    }
  }
}